### ISSUE TITLE
docker not finding root_folder

### ISSUE THREAD

**Issue by shanecasey** [gilesknap/gphotos-sync#187]
Having some trouble getting this working properly with Docker.

Followed the instructions and got it to run properly first time (although I had the timeout issue some other people have had) but when I try and run on docker I get the error
`gphotos-sync: error: the following arguments are required: root_folder`

I did the intitial install through command line on Ubuntu and I'm using Portainer to handle my containers. The yaml is version 2 and looks like
`gphotos-sync:
    image: gilesknap/gphotos-sync
    container_name: gphotos-sync
    volumes:
      - /home/xavier/docker/gphotos:/config
      - /mnt/4904ec68-242c-4325-bde6-58c8c0ee49f1/GPhotos:/storage
    restart: unless-stopped
    environment:
      - PUID=1000
      - PGID=1000`

How do I pass the root_folder argument via docker? I thought the /storage mount was doing that? 

Apologies if I'm doing something stupid here

**Comment by gilesknap** [comment]
Hi Shane,

I'm sorry to say that the docker image was provided by other contributors and I'm not sure of the answer to your question. Hopefully, someone who knows will see this ticket.

Also, I will have a go at running under docker myself so that I can answer yours and future questions. Give me a couple of days to find some time to do so.

**Comment by gilesknap** [comment]
Hopefully, this is the answer to your issue? https://github.com/gilesknap/gphotos-sync/issues/161#issuecomment-564265554

**Comment by shanecasey** [comment]
Unfortunately that's not the answer. I'd read that one.

Small progress: I've found in Portainer's advanced settings for the container that I can add a command. https://i.imgur.com/9zEy6KV.png

So the root_folder is now detected but the logs show the Google authorisation challenge but I can't figure out how to get the response in there. https://i.imgur.com/HckPEmf.png

Will keep plugging away. Maybe will try on Portainer forums as it's maybe me not using that correctly than your issue!

**Comment by gilesknap** [comment]
To do the auth you would need to run interactively the first time (on a separate pc if possible). then you need to get the resulting token file in the correct place to be seen by docker.
I'm working on a couple of bug fixes but after that will have walkthrough of docker deployment if you have not worked it out at that point.

**Comment by shanecasey** [comment]
Seems to be working for me now. 

I was trying to initialise from the stacks composer in Portainer but I deleted the container in Portainer, ran the docker command from the terminal as you described in the readme, and it appears fine in Portainer. I amended the restart policy there and it seems to be running fine now, even after reboots. I think the problem was from following the non-docker setup commands and then trying to create a container. Either way, all seems to be fine now. 

Thanks for writing a super useful tool!
